Bump pxt 12.x and some other changes

THEb0nny opened the pull request:
After switching to pxt-core v10, fieldmotors break.
Related to this.

I understand that these changes should be considered after this
https://github.com/microsoft/pxt-ev3/pull/1048

https://github.com/microsoft/pxt-microbit/pull/5496
https://github.com/microsoft/pxt/pull/9865

THEb0nny committed:
consume-pxteditor-changes-and-upgrade-code

THEb0nny committed:
Update deploy.ts

THEb0nny committed:
pxt-core-and-common-packages-bump

THEb0nny committed:
Update extension.ts

THEb0nny committed:
Update field_music.ts

THEb0nny committed:
Update field_ports.ts

THEb0nny committed:
Update field_music.ts

THEb0nny committed:
Update field_color.ts

THEb0nny committed:
Update field_music.ts

THEb0nny committed:
Update field_music.ts

THEb0nny committed:
Update field_motors.ts

THEb0nny committed:
Update field_brickbuttons.ts

THEb0nny committed:
Update field_motors.ts

https://github.com/microsoft/pxt-microbit/pull/5496

THEb0nny commented:
Fixed all fields field_motors, field_ports, field_brickbuttons, field_music except field_color. When choosing a color, the color does not change. There is no error in the console.

THEb0nny committed:
Update field_ports.ts

THEb0nny committed:
del_field_ports

Replacing ports with images. field ports has been removed. Almost completely repeats the functionality.

THEb0nny committed:
Update field_brickbuttons.ts

THEb0nny commented:
Replacing field ports to images. field ports has been removed. Almost completely repeats the functionality.

All sensor blocks that used field_ports had to register this. I don’t know how it could have been done differently without repeating these lines.
```
this.fieldOptions.columns="4"
this.fieldOptions.width="300"
```

THEb0nny committed:
Update ultrasonic.ts

THEb0nny committed:
Update field_color.ts

THEb0nny committed:
turnRatio-del

Where did turnRatio come from?

THEb0nny committed:
Update field_color.ts

The last change to field_colour removed the addition of className. It’s not clear for what reason, but I’m adding it myself now.

THEb0nny committed:
return-applyColour()

The new version of FieldColorNumber does not apply applyColour() inside the doValueUpdate_() method. Secondly, the applyColour() method was redesigned in the new version so that it no longer worked with this enum code.

THEb0nny commented:
The colorenum field has been fixed.

THEb0nny committed:
bump_10-0-13

THEb0nny committed:
bump-10-0-14

THEb0nny committed:
bump-10-0-16

THEb0nny committed:
bump-10.0.18

THEb0nny commented:
@jwunderl @riknoll can someone see all the changes?

THEb0nny committed:
bump-pxt-10.0.20

THEb0nny committed:
fix-button-select

It turned out that the buttons were wrong for some reason. The sorting of the data in options.data has changed for some reason.

THEb0nny commented:
It turned out that the buttons were wrong for some reason. The sorting of the data in options.data has changed for some reason. You can watch the problem and the solution in the video.

https://github.com/microsoft/pxt-ev3/assets/13646226/dec9d0f1-3fa6-4611-851b-65646a230c6e

THEb0nny committed:
Update field_brickbuttons.ts

THEb0nny committed:
additions-for-downloading-bluetooth

Additional changes that build on this.
https://github.com/microsoft/pxt-ev3/pull/1035
I also wanted to filter by vendorID and productID, but then the selection window did not display anything. I don't know what's wrong.
As before, downloading via bluetooth works on my surface pro 6 with windows 11. Loading did not work on another computer in the robotics class with Windows 10. Perhaps this change will have a positive effect. I will check it.

THEb0nny committed:
add-functionality-and-upgrade -motors-blocks

timeOut is quite an interesting property and if it is used in js, then switching in the block will not be converted.

THEb0nny commented:
For blocks pause until ready and pause until stalled property timeOut is quite an interesting property and if it is used in js, then switching in the block will not be converted.

THEb0nny commented:
The changes that were added to download via bluetooth did not give a good result. I checked on a computer where the download did not work on the previous version. There are also problems with this change. I tried a bunch of different options, and one case worked. The download worked in one of the cases. After that, I wanted to check again, rebooted the computer and started trying bluetooth booting again, but nothing worked.
But I will note that, as before, bluetooth booting worked for my surface pro 6.

On the computer in the classroom the following error pops up, which is not present on my home surface pro 6.
TypeError: Cannot set properties of undefined (setting 'onConnectionChanged')

I will try to solve this separately later.

THEb0nny commented:
On the computer in the classroom the following error pops up, which is not present on my home surface pro 6.
TypeError: Cannot set properties of undefined (setting 'onConnectionChanged')

THEb0nny committed:
bump-pxt-10.0.23

THEb0nny committed:
Update output.ts

THEb0nny committed:
bump-pxt-10.2.3-and-common-packages-10.0.3

THEb0nny committed:
bump-pxt-core-10.2.15

THEb0nny committed:
bump-pxt-10.3.8-and-common-packages-12.1.1

THEb0nny committed:
bump-pxt-11.2.3

THEb0nny committed:
bump_pxt_11.2.10

THEb0nny committed:
downloadArea-delete-fill

THEb0nny commented:
Remove the fill for .collapsedEditorTools #downloadArea
![image](https://github.com/user-attachments/assets/4a8a9dc5-53ee-4bfa-af29-7cc9d311b528)
![image](https://github.com/user-attachments/assets/75573d81-b98f-4832-b80f-85430427c6ad)

THEb0nny committed:
bump-pxt-10.2.15

THEb0nny committed:
Update light.ts

If the light sensor is not active (it is not connected), then do not transfer the value to a new range. Otherwise, the value will be 100.

THEb0nny committed:
suppo-undetectable-analog-sensors

THEb0nny commented:
I made a commit in which I implemented changes for the analog sensor class aimed at supporting older sensors for NXT, which cannot be detected automatically. For example, some Hitechnic sensors.
Here is an example, I got the values from the sensor - https://github.com/THEb0nny/makecode-ev3-hitechnic-eopd-sensor/blob/master/hteopd.ts

THEb0nny committed:
bump_pxt_11.2.20

THEb0nny committed:
bump-pxt-11.3.4-and-common-packages-12.2.1

THEb0nny commented:
@abchatra @riknoll hi, can someone review my changes to finally release a new beta version?

THEb0nny committed:
bump-pxt-11.4.7-and-common-packages-12.3.5

THEb0nny committed:
Update storage-core.ts

THEb0nny committed:
bump-pxt-11.4.28-and-common-packages-12.3.31

THEb0nny committed:
Update field_motors.ts

Fixes for new versions of Blockly, Fixing buttons in showEditor_().
Return scroll otherwise the items won't fit

THEb0nny committed:
Update field_brickbuttons.ts

THEb0nny committed:
Update field_brickbuttons.ts

THEb0nny committed:
Update field_music.ts

Fixes for new versions of Blockly.

THEb0nny committed:
Update storage-core.ts

THEb0nny committed:
Update field_motors.ts, field_music.ts

THEb0nny committed:
Update field_music.ts

THEb0nny committed:
Update field_music.ts

Fix for no voice acting when mouse is hovered. But console shows error:
Uncaught (in promise) AbortError: The play() request was interrupted by a call to pause(). https://goo.gl/LdLk22

THEb0nny commented:
@abchatra @riknoll pr is not ready yet.
So far, support is not completed, without it the appearance is very broken, but I am in the process, although it is difficult for me to figure it out, although I look at other Microbit and Arcade projects.
And there are some mistakes that do not seem to affect the functionality, but are displayed in the console.
For example, there is a simulator crash, I have not reached that yet.

If they gave me some tips how to solve or something was not the best made, it would be great.

THEb0nny commented:
I still have a problem when the assembly is displayed
```Skipped data-uri embedding of ../docs/static/icons/ts-logo-512.svg because file not found```, it does not affect the assembly, but there are a lot of messages of these, I think it affects the assembly speed. I looked in other Microbit and Arcade projects, but there is no such file to take it here to the project.

THEb0nny committed:
theme-support

THEb0nny committed:
Update field_music.ts

There is a problem in pxsim.AudioContextManager, which manifests itself in the fact that if you quickly move the mouse cursor, an error appears in the console:
Uncaught (in promise) AbortError: The play() request was interrupted by a call to pause(). https://goo.gl/LdLk22

THEb0nny committed:
Update field_music.ts

Replacing substr with slice.

THEb0nny commented on editor/wrap.ts:
```diff
@@ -110,14 +110,128 @@ export class Ev3Wrapper {
+            case 0x02:
+                log("Reply Status HANDLE_NOT_READY");
+                break;
+            case 0x03:
```
https://github.com/mindboards/ev3sources/blob/master/lms2012/c_com/source/c_com.h

THEb0nny replied:
I fixed it in the commit below

THEb0nny committed:
Bluetooth log fix and improvements

Optimized the withdrawal of logs about file download. DumpinPutCmd Reply Status should have been by BYTE 6, not 5. Fixed.

THEb0nny commented on editor/deploy.ts:
```diff
@@ -63,26 +64,37 @@ class WebSerialPackageIO implements pxt.packetio.PacketIO {
                     buffer = undefined;
+                } else if (buffer.length > size) {
+                    console.warn(`Received larger bufer then command command: ${buffer.length} recieved but waiting for ${size}`);
+                    let tmp = buffer.slice(0, size - 1);
+                    this.onData(new Uint8Array(tmp));
+                    tmp = buffer.slice(size, buffer.length - 1);
```
I don't remember what this change was connected with, but Bluetooth download didn't break for me.

THEb0nny replied:
I guess you can reject this change. I don't remember why. Downloading still works via Bluetooth. I think I took it from another PR related to the Bluetooth change.

THEb0nny committed:
Update deploy.ts

THEb0nny committed:
allow_underscores_in_filename

Change to allow the use of underscores in file names.

THEb0nny committed:
bump-pxt-11.4.40

THEb0nny committed:
Update blockly.less

Return scroll for default blocklyDropDownContent class.

THEb0nny commented:
Scrolling is missing in the new version of blockly. Scroll y had to be returned.
![image](https://github.com/user-attachments/assets/bb1be2bb-af4e-4fcc-905a-b915ef6128ad)

THEb0nny committed:
changes_in_MMapMethods.buf()_for_sim_recovery

An error that prevented the simulator from starting, citing errors when executing brick.readButtons() and hashDevices(). Simulator crashed, no error handler Error: TypeError: Expected type Buffer but received type object.

Here buf() simply returns an object with a data: Uint8Array field, not a real Buffer (i.e. RefBuffer), which the PXT runtime expects.

The BufferMethods.slice, getNumber, etc. methods expect an object with a Buffer prototype and methods, not a raw JS object { data: Uint8Array }.

By returning a full Buffer from buf() you fix this.

THEb0nny commented:
Fixed simulator crash in last commit.

THEb0nny committed:
bump-pxt-12.1.5-and-common-packages-13.1.0

THEb0nny committed:
bump-pxt-12.2.8-and-common-packages-13.1.4

THEb0nny commented:
I noticed one thing: the EV3 code stops immediately after loading. I don't understand what's causing this. And I don't understand when this happened. I don't remember if I tested it on hardware right away when creating the PR.

@abchatra @riknoll Is this related to the need to update the runtime due to increased pxt-core versions? core---linux?

abchatra commented:
We haven't done any intentional change and don't have bandwidth either to debug.

THEb0nny committed:
Create ts-logo-512.svg

Fixed a build warning:
Skipped data-uri embedding of ../docs/static/icons/ts-logo-512.svg because file not found

THEb0nny committed:
bump-pxt-12.2.20-and-pxt-common-packages-13.1.8

THEb0nny committed:
themes-support-changes

THEb0nny committed:
Merge branch 'master' into bump-pxt-10

THEb0nny commented:
> We haven't done any intentional change and don't have bandwidth either to debug.

I figured out what it was and rolled back the change. It was because I wanted to specify which folder to upload the project to on the controller. I understand that the EV3 bootloader automatically uploads to the folder specified in the firmware. But my file compiles and specifies a different path. This creates two files with incorrect references to each other. I hope it was explained clearly.
I think I did this before, and the project folder was created, but as I understand it now, it happens when uploading via webserial over Bluetooth, where the EV3 bootloader isn't involved.
I want to leave this for future changes and implement USB uploading from webserial over HID, like in the Microbit.

@abchatra @riknoll I've fixed a lot of things in this update. Theme support is now adequate, and the editor's appearance is now fine. There are probably still a few remaining issues, but they're not key, and I'd like to revisit them later after the new beta version is released. All further changes will be made in separate PRs.

First you need to approve this pr because this one works on the basis of that
https://github.com/microsoft/pxt-ev3/pull/1048

Please look at this pr.

THEb0nny committed:
another-style-change

Something was fixed and filemenu was added.

THEb0nny committed:
fullscreen-button change size

THEb0nny committed:
fix neutral btn style

THEb0nny committed:
Update style.less

THEb0nny committed:
fullscreensim-bg-fix-style

THEb0nny committed:
Update deploy.ts

THEb0nny commented on editor/deploy.ts:
```diff
@@ -63,26 +64,37 @@ class WebSerialPackageIO implements pxt.packetio.PacketIO {
+                    console.debug(`Next buffer size: ${this.bufferSize(buffer)}`);
                 } else {
-                    console.warn("Incomplete command " + size);
+                    console.warn(`Incomplete command: ${buffer.length} recieved but waiting for ${size}. Keep waiting...`);
```
Fixed in new commit

THEb0nny commented on libs/storage/storage-core.ts:
```diff
@@ -1,11 +1,11 @@
+    //% block="semicolon"
+    Semicolon
+}
 
-    export enum Separators {
-        //% block="comma"
-        Comma,
-        //% block="semicolon"
-        Semicolon
-    }
+namespace storage {
```
Moved outside the namespace so I don't have to write storage.Separators.Comma.
Or why export it if it's not in the storage namespace? It works without it. I did it using other libraries as examples.

THEb0nny committed:
bluetooth-btn-style-fix

THEb0nny committed:
change-yellow

THEb0nny committed:
downloadArea-style-fix

THEb0nny committed:
secondary-btn-fucus-bg-color-fix

THEb0nny committed:
btn-primary-border-radius-style-upd

THEb0nny committed:
time-machine-styles

THEb0nny committed:
fullscreen-sim-style-fix

THEb0nny committed:
Update style.less

THEb0nny committed:
pxt-focus-border-set-transparent

THEb0nny committed:
move-bluetooth-style-to-light-theme

THEb0nny committed:
blockly-toolbox-style-upd

THEb0nny committed:
download-area-editortools-btn-style

THEb0nny committed:
common-modal-styles

THEb0nny committed:
modal-closeIcon-modal-auto-save-disabled-warning-title-inline-block

THEb0nny committed:
common-extension-card-color-style

THEb0nny committed:
blockly-trash-icon-color

THEb0nny committed:
udp-grid-colour

THEb0nny committed:
fullscreen-modal-styles

THEb0nny committed:
fullscreen-common-modal-header-back-btn-focus-border-none

THEb0nny committed:
downloaddialog-column-hint-bg

THEb0nny committed:
common-card-label-style-udp

THEb0nny committed:
card-hover-i-grey-icon

THEb0nny committed:
features-and-experiments

THEb0nny committed:
field-color-legoColorPicker-fix

THEb0nny committed:
legoColorPicker-color-none-crossed-out

BlocklyDropDownContent legoColorPicker color none crossed out style

THEb0nny commented:
It turned out that the dropdown field-color was slightly broken. I also changed it so that the no-color option would display a red strikethrough.
<img width="348" height="195" alt="image" src="https://github.com/user-attachments/assets/8b6909f7-e574-419a-8147-5898afebeeaa" />

THEb0nny committed:
legoColorPicker-color-none-transparent-bg

THEb0nny commented:
I also made a transparent pattern for the color none in the dropdown.

<img width="337" height="191" alt="image" src="https://github.com/user-attachments/assets/c7625797-330f-4a6b-9b41-451ee616d667" />

THEb0nny committed:
blockly-transparent-pattern-for-colo-none

For color none, I set a transparent pattern instead of a color for the block with colorEnumPicker.

THEb0nny commented:
I also made a transparent pattern for color none in colorEnumPicker.

<img width="494" height="240" alt="image" src="https://github.com/user-attachments/assets/c589206e-82aa-4bef-adfa-ffc6c1b692ee" />

THEb0nny commented:
@abchatra will there be an opportunity to consider changes for the merger?

THEb0nny commented:
@riknoll will there be an opportunity to do a review?

riknoll commented:
@THEb0nny we are no longer updating the EV3 editor. at this point, it can be considered retired. we have no plans currently to remove the hosted website, but unfortunately we don't have the bandwidth to get this repo to the point where it can be deployed again. as such, i'm going to close this PR because we cannot release it.
 
however, if you wish to continue development on your own fork, please see the documentation here for a command in the PXT cli that can be used to produce a static version of the website that you can host with any static webserver (or packaged as an electron app):
 
https://makecode.com/cli/staticpkg
 
please let me know if you have any problems getting setup with statickpkg

THEb0nny commented:
@riknoll @abchatra this sounds very sad. I don't understand.
I thought we reached an agreement a few years ago where I would make changes and the Makecode developers would periodically release new beta versions.
You say the editor is outdated, but I'm making changes so that pxt-ev3 is a modern pxt editor without any input from the Makecode developers. I'm only asking that the Makecode developers verify that the build runs and release the changes in the beta release. Does that really require a lot of resources?

Lego EV3 is still widely used in education...

riknoll commented:
unfortunately, it does require resources on our side to keep our build/release pipelines running, review PRs, test for quality, etc.

as i stated, we have no plans at this time to remove the hosted site. as it stands now the only updates we will consider are those that are necessary to stay compliant with security and privacy obligations.

if you want to release your changes, i recommend hosting your own site or releasing a version on github pages using the staticpkg command.

THEb0nny commented:
@riknoll If I publish a static site, I don't understand how support for string resources running through CrowdIn will work. I also don't understand how sharing a project via a link will work.

riknoll commented:
staticpkg will package all of the existing strings from crowdin, but you won't be able to publish any new ones. i guess if you need updated versions of the common editor strings then you could fetch them by cloning a supported target (like micro:bit) and downloading the translations using the CLI. i don't remember the syntax, but there is a command in the PXT CLI that will fetch all the built translations for a given target.

you are correct that sharing will not work with a static site though

THEb0nny commented:
@riknoll what else might a backend be needed for, besides supporting the project sharing feature? Is it even possible to implement this feature on your own server?

riknoll commented:
it's certainly possible to implement your own sharing API, the API is pretty simple. you might need to modify some settings to make it hit your endpoint, however.

the main other feature our backend provides is C++ compilation, which i'm guessing is not that big a deal for your purposes since i don't think too many people are creating EV3 projects/extensions with C++ in them

THEb0nny commented:
@riknoll I understand that the functions from GitHub don't work either?

THEb0nny commented:
@riknoll How do I use Github Actions to build pxt staticpkg correctly? The build succeeds, but the page returns 404 errors for all files. The URLs listed are /docs/static/..., but they should be /pxt-ev3/docs/static/

Example
https://theb0nny.github.io/docs/static/lessons/use.png

THEb0nny commented:
@riknoll I don't understand, why does the editor in the local version access makecode.com to upload a resource to the galleries on the main page? The editor deployed on the server does the same thing.
For example,
https://www.makecode.com/api/md/ev3/tutorials/infrared-sensor?targetVersion=1.4.41

It's the same as the editor accessing the translation.
For example,
https://makecode.com/api/translations?lang=ru&filename=ev3%2Fev3-jsdoc-strings.json&approved=true
